src/Ledger/Ledger/Properties.agda

```diff
@@ -82,29 +97,611 @@ instance
+    (s : LState)
+    where
+
+    LEDGER-govDepsMatch : ∀ {s' : LState} → Γ ⊢ s ⇀⦇ tx ,LEDGER⦈ s'
+                          → govDepsMatch s → govDepsMatch s'
+
+    LEDGER-govDepsMatch (LEDGER-I⋯ refl (UTXOW-UTXOS (Scripts-No _))) aprioriMatch = aprioriMatch
+
+    LEDGER-govDepsMatch s'@{⟦ utxoSt' , govSt' , certState' ⟧ˡ}
+      utxosts@(LEDGER-V⋯ tx-valid _ _ GOV-sts) aprioriMatch =
+      let
+        open Ledger-sts-setup tx Γ s;  open PParams pp using (govActionDeposit)
+        utxoDeps' = getDeps s'
+        UPD = updateProposalDeposits txprop txid (PParams.govActionDeposit pp) utxoDeps
+        govSt' = LState.govSt s'
+        open SetoidReasoning (≡ᵉ-Setoid{DepositPurpose})
+      in
+      begin
+      filterˢ isGADeposit (dom utxoDeps')
+        ≈⟨ filter-pres-≡ᵉ (≡ᵉ.reflexive (cong dom (STS→utxoDeps≡ utxosts tx-valid))) ⟩
+      filterˢ isGADeposit (dom (updateDeposits pp txb utxoDeps))
+        ≈⟨ noGACerts{txcerts} UPD ⟩
+      filterˢ isGADeposit (dom (updateProposalDeposits txprop txid govActionDeposit utxoDeps))
+        ≈⟨ filter-pres-≡ᵉ (updatePropDeps≡ᵉ{txprop}) ⟩
+      filterˢ isGADeposit (dom (utxoDeps ∪⁺ proposalDepositsΔ txprop pp txb))
+        ≈⟨ filter-pres-≡ᵉ dom∪⁺ ⟩
+      filterˢ isGADeposit ((dom utxoDeps) ∪ dom (proposalDepositsΔ txprop pp txb))
+        ≈⟨ filter-hom-∪ ⟩
+      (filterˢ isGADeposit (dom utxoDeps)) ∪ (filterˢ isGADeposit (dom (proposalDepositsΔ txprop pp txb)))
+        ≈⟨ ∪-cong ≡ᵉ.refl (allGA-propDepsΔ{txprop}) ⟩
+      (filterˢ isGADeposit (dom utxoDeps)) ∪ (dom ((proposalDepositsΔ txprop pp txb)ˢ))
+        ≈⟨ ∪-cong aprioriMatch (utxo-govst-connex{txprop}) ⟩
+      fromList (map f govSt) ∪ fromList (map f (updateGovStates [] (map inj₂ txprop)))
+        ≈˘⟨ ∪-cong ≡ᵉ.refl (≡ᵉ.reflexive (cong fromList (vote-updateGovStates-map-invar++ txvote txprop))) ⟩
+      fromList (map f govSt) ∪ fromList (map f (updateGovStates [] (txgov txb)))
+        ≈⟨ ∪++ ⟩
+      fromList ((map f govSt) ++ (map f (updateGovStates [] (txgov txb))))
+        ≈˘⟨ ≡ᵉ.reflexive (cong fromList (map-++ f govSt (updateGovStates [] (txgov txb)))) ⟩
+      fromList (map f (govSt ++ updateGovStates [] (txgov txb)))
+        ≈˘⟨ ≡ᵉ.reflexive (cong fromList (updateGovStates≡ (txgov txb))) ⟩
+      fromList (map f (updateGovStates govSt (txgov txb)))
+        ≈˘⟨ ≡ᵉ.reflexive (cong (fromList ∘ (map f) ) (STS→GovSt≡ utxosts tx-valid)) ⟩
+      fromList (map f govSt')
+        ∎
+
+
+  -- MAIN THEOREM: EPOCH --
+  module _ (tx : Tx) (let open Tx tx renaming (body to txb)) (let open TxBody txb)
+           (Γ : LEnv) (let open LEnv Γ renaming (pparams to pp))
+           (eΓ : NewEpochEnv)(eps : EpochState)
+    where
+    open EpochState eps hiding (es)
+    open LState ls
+    open GovActionState
+    open RatifyState fut using (removed) renaming (es to esW)
+
+    EPOCH-govDepsMatch :
+      {assumption/bug? : mapˢ (GovActionDeposit ∘ proj₁) removed ⊆ mapˢ proj₁ ((UTxOState.deposits utxoSt)ˢ)}
```

Comment:
This isn't a bug, `removed` should have this property if it's been computed by `RATIFY` in the previous `EPOCH` transition. But of course it might be false when you just make up an arbitrary state. So I think we can think about how to correctly state and prove this in the future.

Reply:
Okay, I'll change the name for now, to something like, ratified-removed.